mlx/backend/metal/metal.h

```diff
@@ -66,4 +66,8 @@ std::function<void()> make_task(
     std::vector<std::shared_future<void>> deps,
     std::shared_ptr<std::promise<void>> p);
 
+/** Capture a GPU trace, saving it to an absolute file `path` */
+bool start_capture(std::optional<std::string> path = std::nullopt);
```

Comment:
The CI fails since `#include <optional>` is missing 
Lets make the following change

```suggestion
bool start_capture(std::string path = "");
```